test/e2e/pipeline/src/test/java/org/apache/shardingsphere/test/e2e/data/pipeline/cases/base/PipelineBaseE2EIT.java

```diff
@@ -383,4 +377,19 @@ protected void assertGreaterThanOrderTableInitRows(final int tableInitRows, fina
         int recordsCount = getTargetTableRecordsCount(tableName);
         assertTrue("actual count " + recordsCount, recordsCount > tableInitRows);
     }
+    
+    protected DataSource generateShardingSphereDataSourceFromProxy() throws SQLException {
+        String dataSourceConfigText = queryForListWithLog("EXPORT DATABASE CONFIGURATION").get(0).get("result").toString();
+        if (PipelineEnvTypeEnum.DOCKER == ENV.getItEnvType()) {
+            DockerStorageContainer storageContainer = ((DockerContainerComposer) containerComposer).getStorageContainers().get(0);
+            String sourceUrl = String.join(":", storageContainer.getNetworkAliases().get(0), Integer.toString(storageContainer.getExposedPort()));
+            String targetUrl = String.join(":", storageContainer.getHost(), Integer.toString(storageContainer.getMappedPort()));
+            dataSourceConfigText = dataSourceConfigText.replace(sourceUrl, targetUrl);
+        }
+        YamlRootConfiguration rootConfig = YamlEngine.unmarshal(dataSourceConfigText, YamlRootConfiguration.class);
+        for (Map<String, Object> each : rootConfig.getDataSources().values()) {
```

Comment:
`each` should be `entry`, refer to Code of Conduct for more details

Reply:
It's `values`, not `entrySet()`, do we need use `entry`?

Reply:
OK, I had a mistake